mkosi/__init__.py

```diff
@@ -6505,7 +6513,15 @@ def load_args(args: argparse.Namespace) -> MkosiArgs:
     # runs. We'll strip it off the final image, but some build-time
     # tools (dracut...) want a fixed one, hence provide one, and
     # always the same
-    args.machine_id = uuid.uuid4().hex
+    if args.machine_id is None:
+        args.machine_id = uuid.uuid4().hex
+        args.machine_id_is_fixed = False
+    else:
+        args.machine_id_is_fixed = True
+        try:
+            uuid.UUID(hex=args.machine_id)
+        except:
```

Comment:
This should be `except Exception`

Reply:
Ah, you were a bit quicker than me. :) I think something more specific than `Exception` would better, though.